### PR TITLE
test images: Updates Image Builder process in the README

### PR DESCRIPTION
**Pull request by claudiubelu**
**What type of PR is this?**

/kind documentation

/sig testing
/sig windows

**What this PR does / why we need it**:

A few details about the image builder postsubmit jobs changed, so the README had to be updated as well.

Added a few extra bits of information regarding the Windows images.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by aojea** [comment]
/lgtm
for the image buiolding README
who should approve the windows docs?

**Comment by claudiubelu** [comment]
/asign @marosset 

FYI, the windows/README.md file is pretty similar to what we had before the docker buildx switch.

File: test/images/README.md [hunk 1]
```diff
@@ -105,6 +112,12 @@ limitations when it comes to the Windows images:
   (make sure that process isolation is used, not Hyper-V isolation).
   [This](https://stefanscherer.github.io/find-dependencies-in-windows-containers/) is a useful guide on how to use `procmon.exe`.
 
+For Windows, in order to spawn process-isolated containers, the container OS version should closely match
+the host OS version. For this reason, we build test images for different Windows OS Versions: 1809, 1903,
+1909. In order add support for a new Windows OS version, a new entry for that OS version will have to
+be first added to the `windows-servercore-cache` and `busybox` images, followed by the rest of the images.
```
Comment (jsturtevant) [inline test/images/README.md:118]: Can we provide a link to test/images/windows/README.md here?

File: test/images/README.md [hunk 2]
```diff
@@ -54,26 +54,33 @@ to build the image and push it to your own registry first, and run some tests th
...
+
 The images are built through `make`. Since some images (e.g.: `busybox`) are used as a base for
 other images, it is recommended to build them first, if needed.
 
 
+### Creating and promoting new images
+
+If you intend to add an entirely different image and have it automatically built by the Image Builder
+and used in E2E tests, you will also have to define the postsubmit prow job for it. This can easily
+be done by running [this script](https://github.com/kubernetes/test-infra/blob/10a290aaa9f2657ec46d8a2a50e0a3eeca5b1715/config/jobs/image-pushing/k8s-staging-e2e-test-images.sh)
```
Comment (jsturtevant) [inline test/images/README.md:80]: other links are set to the `master` branch should this one be linked to that as well?

File: test/images/README.md (hunk 1 above)
Comment (jsturtevant) [reply test/images/README.md:118]: The doc talks about how to build your own.  Where do the official ones live and how do publish for use here?  Can we add a section for that to the readme below?

Comment (claudiubelu) [reply test/images/README.md:118]: For ``windows-servercore-cache`` and ``busybox`` images, this same README file applies. They are built the same way all other regular test image is built. Don't mistake them for the ``busybox-helper`` and ``powershell-helper`` images.

Comment (claudiubelu) [reply test/images/README.md:118]: As for the second question, I'm not sure I understand what you're referring to. Do you mean the Image Promotion process / official E2E test registry? If so, it's already been covered in the other sections of this README.

File: test/images/windows/README.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+az vm open-port -g GROUP-NAME -n NODE-NAME --port 2376
+```
+
+The `ca.pem`, `cert.pem`, and `key.pem` files that can be found in `$env:USERPROFILE\.docker`
+will have to copied to the `~/.docker/` on the Linux build node.
+
+```powershell
+scp.exe $env:USERPROFILE\.docker\*.pem ubuntu@YOUR_LINUX_BUILD_NODE:/home/ubuntu/.docker/
+-After all this, the Linux build node should be able to connect to the Windows build node:
+
```
Comment (marosset) [inline test/images/windows/README.md:60]: nit: missing ``` to close the ```powershell snippet

Comment (claudiubelu) [reply test/images/windows/README.md:60]: Done

**Comment by marosset** [comment]
the windows changes LGTM (except for a formatting nit I commented on)

File: test/images/README.md (hunk 1 above)
Comment (jsturtevant) [reply test/images/README.md]: I think I got them confused with `busybox-helper` and `powershell-helper`.  Are there docs on how those are built and published?

Comment (marosset) [reply test/images/README.md]: > I think I got them confused with busybox-helper and powershell-helper. Are there docs on how those are built and published?

This PR adds a readme to /test/images/windows/ which is where the helper images get built.

**Comment by dims** [comment]
/approve
/lgtm

thanks for the detailed reviews on this @marosset @jsturtevant

**Pull request merged**